### PR TITLE
feat: add city, and distance on kilometric expense line card

### PR DESCRIPTION
**Pull request by gca-axelor**
RM#72150

### CODE REVIEW

**Review by vhu-axelor — CHANGES_REQUESTED** [review]

File: packages/apps/hr/src/i18n/en.json [hunk 1]
```diff
@@ -42,6 +42,8 @@
   "Hr_Refuse": "Refuse",
   "Hr_ReasonRefusal": "Reason for refusal",
   "Hr_CityNotFound": "At least one of the two cities could not be located.",
+  "Hr_KilometerUnit": "km",
+
```
Comment (lme-axelor) [inline packages/apps/hr/src/i18n/en.json:46]: remove space, not needed in translations

File: packages/apps/hr/src/i18n/fr.json [hunk 2]
```diff
@@ -42,6 +42,8 @@
   "Hr_Refuse": "Refuser",
   "Hr_ReasonRefusal": "Motif de refus",
   "Hr_CityNotFound": "Au moins une des deux villes n'a pas pu être localisée.",
+  "Hr_KilometerUnit": "km",
+
```
Comment (lme-axelor) [inline packages/apps/hr/src/i18n/fr.json:46]: remove space, not needed in translations

File: packages/ui/src/components/templates/ObjectCard/ObjectCard.tsx [hunk 3]
```diff
@@ -165,6 +159,12 @@ const ObjectCard = ({
       return null;
     }
 
+    if (item.customComponent != null) {
+      return React.cloneElement(item.customComponent, {
+        key: `${item.displayText} - ${item.order}`,
+      });
+    }
+
```
Comment (lme-axelor) [inline packages/ui/src/components/templates/ObjectCard/ObjectCard.tsx:167]: I'm not sure it's a good idea to move it this way. 
I understand the need of having the hideIf being evaluate as a first thing but the hideIfNull can cause issue with the custom component as displayText and indicatorText are not used. Maybe we can split condition into two parts like this : 

```
  if (item.hideIf) {
      return null;
    }

    if (item.customComponent != null) {
      return React.cloneElement(item.customComponent, {
        key: `${item.displayText} - ${item.order}`,
      });
    }

  if (item.hideIfNull &&
        checkNullString(item.displayText) &&
        checkNullString(item.indicatorText)) {
      return null;
    }
```

**Review by vhu-axelor — APPROVED** [review]

**Comment by lme-axelor** [comment]
Need to be splited into two commits before merging to go down to version 7.1

**Pull request merged**